### ISSUE TITLE
kstring.h error: initializer-string for array of chars is too long

### ISSUE THREAD

**Issue by cwrussell** [samtools/htslib#771]
When trying to install another program (pindel), there is an error related to kstring.h

```
make -C src pindel
make[1]: Entering directory '/mnt/projects/cwrussell/MAE/svre/pindel/Pindel/src'
g++  -I/mnt/projects/cwrussell/MAE/svre/pindel/htslib -Wall -g -c -O3 -fopenmp pindel.cpp -o pindel.o
In file included from /mnt/projects/cwrussell/MAE/svre/pindel/htslib/htslib/khash.h:131:0,
    from pindel.h:34,
    from pindel.cpp:35:
/mnt/projects/cwrussell/MAE/svre/pindel/htslib/htslib/kstring.h: In function ‘int kputuw(unsigned int, kstring_t*)’:
/mnt/projects/cwrussell/MAE/svre/pindel/htslib/htslib/kstring.h:221:9: error: initializer-string for array of chars is too long [-fpermissive]
         "90919293949596979899";
         ^
```

Changing line 211 in kstring.h from `static const char kputuw_dig2r[200] =` to `static const char kputuw_dig2r[201] =` allows the pindel installation to occur without error, although I'm unsure of any other consequences related to this change.

**Comment by daviesrob** [comment]
Yes, fixing `kputuw_dig2r` is fairly simple, but when I tried to build pindel after that I got problems with `SIZE_MAX` not being defined.  It appears you don't get `SIZE_MAX` in c++ without defining `__STDC_LIMIT_MACROS` before including `stdint.h`.  Also there's no guarantee that `kstring.h` was the first file to include it (in fact when building pindel, it isn't).

I'm currently trying to brew up a suitable solution to that one that isn't too messy.  Unfortunately the c++ equivalent `std::numeric_limits<size_t>::max()` isn't a constexpr until c++-11, so not terribly ideal.

**Comment by daviesrob** [comment]
Hmmm, `#define SIZE_MAX static_cast<size_t>(-1)` looks like it might be a winner.
